git-ftp

```diff
@@ -845,6 +851,20 @@ set_remote_password() {
 	[ -z "$REMOTE_PASSWD" ] && REMOTE_PASSWD="$(get_config password)"
 }
 
+set_branch() {
+    [ -z $BRANCH ] && BRANCH="$(get_config branch)"
+    if [ ! -z $BRANCH ]; then
+	CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
```

Comment:
This doesn't work if you have a detached HEAD, for example you are on origin/master. I solved it this way: https://github.com/git-ftp/git-ftp/blob/pull/git-ftp#L986